src/test/shell/bazel/remote/remote_execution_test.sh

```diff
@@ -1157,6 +1157,46 @@ EOF
   || fail "Expected bazel-bin/a/template.txt to be downloaded"
 }
 
+function test_downloads_minimal_hit_action_cache() {
+  # Test that remote metadata is saved and action cache is hit across server restarts when using
+  # --remote_download_minimal
+  mkdir -p a
+  cat > a/BUILD <<'EOF'
+genrule(
+  name = "foo",
+  srcs = [],
+  outs = ["foo.txt"],
+  cmd = "echo \"foo\" > \"$@\"",
+)
+
+genrule(
+  name = "foobar",
+  srcs = [":foo"],
+  outs = ["foobar.txt"],
+  cmd = "cat $(location :foo) > \"$@\" && echo \"bar\" >> \"$@\"",
+)
+EOF
+
+  bazel build \
+    --experimental_action_cache_store_output_metadata \
+    --remote_executor=grpc://localhost:${worker_port} \
+    --remote_download_minimal \
+    //a:foobar || fail "Failed to build //a:foobar"
+
+  (! [[ -f bazel-bin/a/foo.txt ]] && ! [[ -f bazel-bin/a/foobar.txt ]]) \
```

Comment:
`bazel-bin/a` will be created by [`SkyframeActionExecutor`](https://github.com/bazelbuild/bazel/blob/a52b1320da06c85d5aaa1ece9eb46acabd09b33f/src/main/java/com/google/devtools/build/lib/skyframe/SkyframeActionExecutor.java#L1005) since it's the output directory.

Updated the code to check `bazel-bin/a` is empty.